cachito/workers/tasks/utils.py

```diff
@@ -97,3 +105,57 @@ def _assert(
             relpath = fullpath.relative_to(self._root_dir)
             err_msg = err_template.format(relpath=relpath)
             raise ValidationError(f"File check failed for {self._pkg_manager}: {err_msg}")
+
+
+def runs_if_request_in_progress(task_fn):
+    """
+    Decorate a task to make it check request state before proceeding.
+
+    If request state is not "in_progress", the task will log an error and exit.
+
+    :param task task_fn: the task to run if its state is in_progress
+    """
+
+    @functools.wraps(task_fn)
+    def task_with_state_check(*args, **kwargs):
+        """Check request state, proceed to execute task if check succeeds."""
+        request_id = get_function_arg_value("request_id", task_fn, args, kwargs)
+        if not request_id:
+            raise ValueError(
+                f"Failed during state check: no request_id found for {task_fn.__name__} task",
+            )
+        request_state = get_request_state(request_id)
+        if request_state != "in_progress":
+            log.error(
+                "Skipping %s task because the request was not in_progress (was: %s)",
+                task_fn.__name__,
+                request_state,
+            )
+            return
+
+        return task_fn(*args, **kwargs)
+
+    return task_with_state_check
+
+
+def get_request_state(request_id):
+    """
+    Get the state of the request.
+
+    :param int request_id: the Cachito request ID this is for
+    """
+    from cachito.workers.requests import requests_auth_session
+
+    config = get_worker_config()
+    request_url = f'{config.cachito_api_url.rstrip("/")}/requests/{request_id}'
+
+    rv = requests_auth_session.get(request_url)
+
+    try:
+        rv.raise_for_status()
+    except requests.RequestException:
```

Comment:
The `get()` method should be called in the try/except block.

`get()` will raise an exception if the connection fails. `raise_for_status()` will raise an exception if the connection works fine but the status code in the response is between 400 and 599.